frontend/src/views/Lobby.tsx

```diff
@@ -1,41 +1,47 @@
 import React, { useCallback, useEffect, useState } from 'react';
-import { useLocation } from 'react-router-dom';
+import { useLocation, Redirect } from 'react-router-dom';
 import { Message } from 'stompjs';
-import { LargeText, UserNicknameText } from '../components/core/Text';
-import {
-  addUser, deleteUser, connect, routes, subscribe, User,
-} from '../api/Socket';
 import ErrorMessage from '../components/core/Error';
+import { LargeText, UserNicknameText } from '../components/core/Text';
+import { connect, routes, subscribe } from '../api/Socket';
+import { Room } from '../api/Room';
+import { User } from '../api/User';
 
 type LobbyPageLocation = {
-  nickname: string;
-}
+  user: User,
+  room: Room,
+};
 
 function LobbyPage() {
   const location = useLocation<LobbyPageLocation>();
 
   // Set the nickname variable.
-  const [nickname, setNickname] = useState('');
+  const [user, setUser] = useState<User | null>(null);
 
   // Hold error text.
   const [error, setError] = useState('');
 
+  // Variable to determine whether to redirect back to join page
+  const [shouldRedirect, setShouldRedirect] = useState(false);
```

Comment:
Right now, there's a brief moment where you are on the lobby before you are redirected back. Am I correct in assuming that the next PR will take care of the actual room validation?

If that is true, then maybe you don't not need all this redirect logic, and instead you could just throw an error? Since you'll end up deleting this after merging #43, if I understand correctly.

Reply:
As for flashing, yeah I think that might be one of the drawbacks for using redirect (I'm not sure if `history.push` has the same issue or not). As for the second part, I'm not too sure - I was assuming that people who navigated to `/game/lobby` but hadn't actually joined/created a game (i.e. arrived at the lobby from a join/create page redirect) would be redirected back to the join page, however I'm totally open to changing the behavior if we all had something else in mind

Reply:
I think in the future, there could be a brief "loading" state which the user sees, and then is redirected to a page with the message "you need to provide a valid room ID to enter the lobby". So this is definitely a fixable issue, and I think it is totally fine for now.